### ISSUE TITLE
Font Library: Make the delete confirm dialog close immediatly after clicking 'Delete'

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#58386]
### Description

When deleting an installed fonr and clicking 'Delete' in the confirm dialog, the confirm dialog stays open untile the font is actually deleted and the fonr library is refreshed.

This seems to be less than ideal for a few reasons>
- As the dialog stays open, users can click 'Delete' ultiple times, especially on a slow internet.
- By clicking multiple times, a new delete request is triggered, which leads to errors like the one in the screenshot below:

![Screenshot 2024-01-29 at 14 03 07](https://github.com/WordPress/gutenberg/assets/1682452/a2cf6cc9-dc6d-438c-b767-3bc78ccf0d38)

- Whatever teherror is, for example no internet connection, the confirm dialog stays open and the error notice appears behind the confirm dialog overlay. The actual error message is partially hidden. Depending on the size of the screen in use, the error may even be completely hidden. I'm not sure hiding error messages is a good user experience. Screenshot:

![Screenshot 2024-01-29 at 15 10 53](https://github.com/WordPress/gutenberg/assets/1682452/85d9963c-7eb9-40ea-b949-6d8359583d65)

### Step-by-step reproduction instructions

- Go to Site editor > Styles > Typography > Manage fonts.
- Install a Google font.
- Go to the Library tab and click the font you installed.
- Click 'Delete' at the bottom of the modal dialog. A confirm dialog opens.
- Quickly click 'Delete' multiple times.
- Observe an error message appears within the Fonts modal dialog, behind the confirm dialog overlay.
- Observe that after the font is actually deleted, the confirm dialog and the notice with the error messsage disappear.
- Note: for accessibility, when the confirm dialog closes, focus should be better handled. This is tracked in another issue.
- Install a new Google font.
- Go to the Library tab and click the font you installed.
- In your browser dev tools > Network, set your browseer to 'offline'.
- Click Delete and then Delete again in the confirm dialog.
- Observe the confirm dialog stays open.
- Observe an error message appears within the Fonts modal dialog, behind the confirm dialog overlay.
- Click Delete again. Observe the confirm dialog stays always open.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
In a first moment I was thinking the simplest fix would be to set `aria-disabled` on the Delete button and `noop` it after the first click. That would prevent users from clicking more than once but it's not ideal. The confirm dialog would still stay open, fora fraction of second or indefinitely depending on the error type.

Thinking bdtter at this, I think that it's not expected for a confirm dialog to stay open i the first place. The very nature of a confirm dialog is to ask for a Yes or No. Once that answer has been given by the user, the confirm dialog should close immediately. Any other feedback like progress of the pending action, success or error, should be provided in the main UI and certainly not by keeping the confirm dialog open.

**Comment by matiasbenedetto** [comment]
> Any other feedback like progress of the pending action, success or error, should be provided in the main UI and certainly not by keeping the confirm dialog open.

It seems in line with the changes added in https://github.com/WordPress/gutenberg/pull/59036

https://github.com/WordPress/gutenberg/assets/1310626/2fc51d84-13ef-4424-a0de-7478cff352f3
